compiler/optimizer/Inliner.cpp

```diff
@@ -5518,9 +5518,11 @@ bool TR_InlinerBase::inlineCallTarget2(TR_CallStack * callStack, TR_CallTarget *
    int32_t bytecodeSize = getPolicy()->getInitialBytecodeSize(calleeSymbol->getResolvedMethod(), calleeSymbol, comp());
    TR_ResolvedMethod * reportCalleeMethod = calleeSymbol->getResolvedMethod();
    int32_t numberOfLocalsInCallee = reportCalleeMethod->numberOfParameterSlots() + reportCalleeMethod->numberOfTemps();
-   TR::DebugCounter::prependDebugCounter(comp(), "inliner.callSites/succeeded", callNodeTreeTop);
-   TR::DebugCounter::prependDebugCounter(comp(), "inliner.callSites/succeeded:#bytecodeSize", callNodeTreeTop, bytecodeSize);
-   TR::DebugCounter::prependDebugCounter(comp(), "inliner.callSites/succeeded:#localsInCallee", callNodeTreeTop, numberOfLocalsInCallee);
+
+   // The debug counters below causes decrement reference count exception during optimization phase
+   // TR::DebugCounter::prependDebugCounter(comp(), "inliner.callSites/succeeded", callNodeTreeTop);
+   // TR::DebugCounter::prependDebugCounter(comp(), "inliner.callSites/succeeded:#bytecodeSize", callNodeTreeTop, bytecodeSize);
+   // TR::DebugCounter::prependDebugCounter(comp(), "inliner.callSites/succeeded:#localsInCallee", callNodeTreeTop, numberOfLocalsInCallee);
```

Comment:
I don't believe these counters will work on other platforms. My tests have shown that when I enable just one of them,  it will cause an inconsistency in the trees during IL optimization, resulting in a reference count error. Running a personal build on the other platforms with these counters enabled results in the same symptoms. I think this is a _common_ issue.